### PR TITLE
Fix flaky connection metric test

### PR DESCRIPTION
**Pull request by assafad**
### What this PR does
Remove any synchronization assumptions from `kubevirt_vmi_last_api_connection_timestamp_seconds` e2e test, which introduced flakiness. Change the test to only check that the metric value gets increased when a connection is being made.

Before this PR:
`kubevirt_vmi_last_api_connection_timestamp_seconds` e2e test was flaky.
After this PR:
`kubevirt_vmi_last_api_connection_timestamp_seconds` e2e test is no longer flaky.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/monitoring/vm_monitoring.go [hunk 1]
```diff
@@ -278,14 +278,28 @@ var _ = Describe("[Serial][sig-monitoring]VM Monitoring", Serial, decorators.Sig
...
+			fmt.Print(secondaryTimestamp)
+			fmt.Print("\n")
+			time.Sleep(30 * time.Second)
 			Expect(console.LoginToAlpine(vmi)).To(Succeed())
 			secondaryMetricValue := validateLastConnectionMetricValue(vmi, secondaryTimestamp)
+			fmt.Print("secondaryMetricValue\n")
+			fmt.Print(secondaryMetricValue)
+			fmt.Print("\n")
 
 			Expect(secondaryMetricValue).To(BeNumerically(">", initialMetricValue))
```
[the remark this answers is not in the document]
Comment (assafad) [reply tests/monitoring/vm_monitoring.go:304]: With this we would only check that the metric gets increased, and wouldn't check that its value is correct.
I think we should keep the checks against `time.Now().Unix()` if possible.

[the remark this answers is not in the document]
Comment (machadovilaca) [reply tests/monitoring/vm_monitoring.go:304]: `validateLastConnectionMetricValue` validates `>=`, not that the value is correct
not much to gain except possible weird results, time synchronization is never guaranteed

maybe you can get the value and validate that corresponds to a timestamp in the last X minutes

initalLocalTime - 60seconds < METRIC_VALUE < initalLocalTime + 60seconds

[the remark this answers is not in the document]
Comment (assafad) [reply tests/monitoring/vm_monitoring.go]: updated, all synchronization assumption was removed. Not only checking the value gets increased.

**Comment by machadovilaca** [comment]
/lgtm

i think the test can still be improved, but lets discuss it later, to unblock this and fix the flakiness

**Review by sradco — APPROVED** [review]

**Pull request merged**